### PR TITLE
random stuff

### PR DESCRIPTION
**Pull request by TesterWoWDev**
Functions I needed.

### CODE REVIEW

**Comment by TesterWoWDev** [comment]
should maybe wrap the MoveTo finalAngle in an Optional<> as it does later down the chain, or make a method overload.

File: tswow-core/Public/TSInstance.h [hunk 1]
```diff
@@ -46,6 +46,7 @@ class TC_GAME_API TSInstance : public TSMap {
     TSInstance* operator->() { return this; }
     operator bool() const { return map != nullptr && m_script != nullptr; }
     bool IsNull();
+    void RemoveFromMap(TSPlayer player, bool remove);
```
Comment (ihm-tswow) [inline tswow-core/Public/TSInstance.h]: should remove have a default value here?

File: tswow-core/Public/TSUnit.h [hunk 2]
```diff
@@ -192,7 +192,8 @@ class TC_GAME_API TSUnit : public TSWorldObject {
     void MoveChase(TSUnit target, float dist, float angle);
     void MoveConfused();
     void MoveFleeing(TSUnit target, uint32 time);
-    void MoveTo(uint32 id, float x, float y, float z, bool genPath);
+    void MoveTo(uint32 id, float x, float y, float z, bool genPath, float finalAngle);
```
Comment (ihm-tswow) [inline tswow-core/Public/TSUnit.h]: finalAngle should be optional

File: tswow-core/Public/global.d.ts [hunk 3]
```diff
@@ -177,6 +177,17 @@ declare const enum DeathStatus {
...
+declare const enum LineOfSightChecks {
+    LINEOFSIGHT_CHECK_VMAP      = 0x1, // check static floor layout data
+    LINEOFSIGHT_CHECK_GOBJECT   = 0x2, // check dynamic game object data
+    LINEOFSIGHT_ALL_CHECKS      = (LINEOFSIGHT_CHECK_VMAP | LINEOFSIGHT_CHECK_GOBJECT)
+}
+
+declare const enum VMapModelIgnoreFlags {
+    Nothing = 0x00,
+    M2      = 0x01
+}
```
Comment (ihm-tswow) [inline tswow-core/Public/global.d.ts]: check if you can read these automatically from trinitys source code, you can look at the other enum declarations at the top of the file for reference. essentially you use syntax like this:

```ts
declare const enum LineOfSightChecks {} /** File.h:TrinityEnumName */
```

File: tswow-core/Public/TSInstance.h (hunk 1 above)
Comment (TesterWoWDev) [reply tswow-core/Public/TSInstance.h]: Everywhere I saw it used in map, they always gave a bool value for it. As it's a call for DeleteFromWorld, not to just remove from the map.

File: tswow-core/Private/TSInstance.cpp [hunk 4]
```diff
@@ -16,6 +16,11 @@ bool TSInstance::IsNull()
     return m_script == nullptr || map == nullptr;
 }
 
+void TSInstance::RemoveFromMap(TSPlayer player, bool remove)
+{
+    map->RemovePlayerFromMap(player.player, true);
+}
```
Comment (ihm-tswow) [inline tswow-core/Private/TSInstance.cpp]: doesn't use the "remove" variable

Comment (TesterWoWDev) [reply tswow-core/Private/TSInstance.cpp]: I never said i was the smartest today.

File: tswow-core/Public/TSInstance.h (hunk 1 above)
Comment (ihm-tswow) [reply tswow-core/Public/TSInstance.h]: that's right, maybe we can call it `deleteFromWorld` to make that more clear

**Comment by TesterWoWDev** [comment]
updated commit #522a41b

**Pull request merged**